akka-javasdk/src/main/java/akka/javasdk/consumer/MessageContext.java

```diff
@@ -22,4 +21,21 @@ public interface MessageContext extends MetadataContext {
 
   /** Access to tracing for custom app specific tracing. */
   Tracing tracing();
+  /**
+   * Returns true if this message was originated on the same region that is being processed.
```

Comment:
maybe clarify what we mean by "originated"? The region where the event was created.

btw, is it "originated on the same region" or "originated from the same region"? below we also have "originated in the region" 😄